examples/train-text-from-scratch/train-text-from-scratch.cpp

```diff
@@ -302,9 +302,9 @@ void init_model(struct my_llama_model * model) {
         // ggml_set_name(layer.w2, (layers_i + ".feed_forward.w2.weight").c_str());
         // ggml_set_name(layer.w3, (layers_i + ".feed_forward.w3.weight").c_str());
 
-        strncpy(layer.w1->name, (layers_i + ".feed_forward.w1.weight").c_str(), sizeof(layer.w1->name));
-        strncpy(layer.w2->name, (layers_i + ".feed_forward.w2.weight").c_str(), sizeof(layer.w2->name));
-        strncpy(layer.w3->name, (layers_i + ".feed_forward.w3.weight").c_str(), sizeof(layer.w3->name));
+        strncpy(layer.w1->name, (layers_i + ".feed_forward.w1.weight").c_str(), sizeof(layer.w1->name) - 1);
+        strncpy(layer.w2->name, (layers_i + ".feed_forward.w2.weight").c_str(), sizeof(layer.w2->name) - 1);
+        strncpy(layer.w3->name, (layers_i + ".feed_forward.w3.weight").c_str(), sizeof(layer.w3->name) - 1);
```

Comment:
This change won't work - see the comment above the lines

Reply:
I would suggest using the recently added `ggml_format_name` here instead.

Reply:
If the name buffers are too small, let's increase the size instead of hacking non-null terminated strings.

Reply:
I came back to report that I switched to `snprintf()` before committing and saw this. Won't need the string formatting capability but it looks like it saves characters